Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -413,16 +429,33 @@ void TextureMapperLayer::collectDamageSelf(TextureMapperPaintOptions& options, D
     m_damage = Damage();
 }
 
+void TextureMapperLayer::damageWholeLayerDueToTransformChangeIfNeeded(const TransformationMatrix& beforeChange, const TransformationMatrix& afterChange)
+{
+    // When the layer's transform changes, we must not only damage whole layer using new transform,
+    // but also using old transform to cover the area not affected by layer anymore.
+    if (beforeChange != afterChange && !m_damage.isInvalid()) {
```

Comment:
Do an early return here, or move this check to the caller and remove the IfNeeded since this is only caled in one place. I would also check if damage is invalid first, before comparing the matrix.